### PR TITLE
feat(github): add github_app id + private key values

### PR DESCRIPTION
**Pull request by JeanBarriere**
Should we include the GitHub Dev app values into the CI for our dev platforms ?

https://github.com/organizations/storyscript/settings/apps/storyscript-development

-> values are stored into lastpass already

### CODE REVIEW

File: templates/runtime/runtime.deployment.yaml [hunk 1]
```diff
@@ -37,3 +37,13 @@ spec:
                 secretKeyRef:
                   name: "{{ .Release.Name }}-runtime"
                   key: wolfram_app_id
+            - name: GITHUB_APP_ID
```
Comment (beardedtim) [inline templates/runtime/runtime.deployment.yaml:40]: I wonder:

I know _eventually_ the goal is to use the _end user's_ github token. We also have the ability to use their token in the current implementation that you built. Should we even include the ability to use ours or should we figure out the OAuth flow and get a new token every story?

Comment (JeanBarriere) [reply templates/runtime/runtime.deployment.yaml:40]: As we already use the GitHub app for the auth flow, I'd say we should try to keep it simple with it and let it be the same for the app used.
@anukul Should we add default values here ?

**Comment by JeanBarriere** [comment]
@anukul @williammartin I think we should add these values for the dev envs from here:
https://github.com/storyscriptinc/ci/blob/master/common-tasks/generate-helm-values.sh#L18-L19

But I don't know where it gets those values

**Comment by JeanBarriere** [comment]
@williammartin @anukul we need to find a fix here as some of this feature is already approved

**Comment by williammartin** [comment]
> a fix here 

A fix for what?

**Comment by JeanBarriere** [comment]
inserting the github app credentials

File: templates/runtime/runtime.secrets.yml [hunk 2]
```diff
@@ -5,3 +5,5 @@ metadata:
 type: Opaque
 data:
   wolfram_app_id: {{ required "A .Values.runtime.wolfram_app_id is required" .Values.runtime.wolfram_app_id | b64enc }}
+  github_app_id: {{ required "A .Values.runtime.github_app.id is required" .Values.runtime.github_app.id | b64enc }}
+  github_app_private_Key: {{ required "A .Values.runtime.github_app.private_key is required" .Values.runtime.github_app.private_key | b64enc }}
```
Comment (williammartin) [inline templates/runtime/runtime.secrets.yml:9]: `Key` should probably be lower case here?

File: templates/runtime/runtime.deployment.yaml [hunk 3]
```diff
@@ -37,3 +37,13 @@ spec:
...
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-runtime"
+                  key: github_app_id
+            - name: GITHUB_APP_PRIVATE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-runtime"
+                  key: github_app_private_Key
```
Comment (williammartin) [inline templates/runtime/runtime.deployment.yaml:49]: `Key` should probably be lower case here?

File: values.yaml [hunk 4]
```diff
@@ -104,6 +104,9 @@ runtime:
     repository: storyscript/runtime
     tag: v0.6.0
   wolfram_app_id:
+  github_app:
+    id:
+    private_key:
```
Comment (anukul) [inline values.yaml:109]: Should we call this `github.app_id` and `github.private_key`, similar to how the key `github` is used in other places, e.g. `creds`?

Comment (JeanBarriere) [reply values.yaml:109]: Ohhh, true, my bad !

File: templates/runtime/runtime.deployment.yaml (hunk 3 above)
Comment (JeanBarriere) [reply templates/runtime/runtime.deployment.yaml]: :+1: done

**Comment by JeanBarriere** [comment]
updated naming and fixed typo @williammartin @anukul :+1:

**Pull request merged**